opennlp-tools/src/main/java/opennlp/tools/ml/model/ComparableEvent.java

```diff
@@ -41,45 +42,75 @@ public ComparableEvent(int oc, int[] pids) {
 
   public int compareTo(ComparableEvent ce) {
 
-    if (outcome < ce.outcome)
-      return -1;
-    else if (outcome > ce.outcome)
-      return 1;
+    int compareOutcome = Integer.compare(outcome, ce.outcome);
+    if (compareOutcome != 0) {
+      return compareOutcome;
+    }
 
-    int smallerLength = predIndexes.length > ce.predIndexes.length ? ce.predIndexes.length
-        : predIndexes.length;
+    int smallerLength = Math.min(predIndexes.length, ce.predIndexes.length);
 
     for (int i = 0; i < smallerLength; i++) {
-      if (predIndexes[i] < ce.predIndexes[i])
-        return -1;
-      else if (predIndexes[i] > ce.predIndexes[i])
-        return 1;
+      int comparePredIndexes = Integer.compare(predIndexes[i], ce.predIndexes[i]);
+      if (comparePredIndexes != 0) {
+        return comparePredIndexes;
+      }
       if (values != null && ce.values != null) {
-        if (values[i] < ce.values[i])
-          return -1;
-        else if (values[i] > ce.values[i])
-          return 1;
+        float compareValues = Float.compare(values[i], ce.values[i]);
+        if (!Float.valueOf(compareValues).equals(Float.valueOf(0.0f))) {
+          return (int) compareValues;
+        }
       } else if (values != null) {
-        if (values[i] < 1)
-          return -1;
-        else if (values[i] > 1)
-          return 1;
+        float compareValues = Float.compare(values[i], 1.0f);
+        if (!Float.valueOf(compareValues).equals(Float.valueOf(0.0f))) {
+          return (int) compareValues;
+        }
       } else if (ce.values != null) {
-        if (1 < ce.values[i])
-          return -1;
-        else if (1 > ce.values[i])
-          return 1;
+        float compareValues = Float.compare(1.0f, ce.values[i]);
+        if (!Float.valueOf(compareValues).equals(Float.valueOf(0.0f))) {
+          return (int) compareValues;
+        }
       }
     }
 
-    if (predIndexes.length < ce.predIndexes.length)
-      return -1;
-    else if (predIndexes.length > ce.predIndexes.length)
-      return 1;
+    int comparePredIndexesLength = Integer.compare(predIndexes.length, ce.predIndexes.length);
+    if (comparePredIndexesLength != 0) {
+      return comparePredIndexesLength;
+    }
 
     return 0;
   }
 
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj)
+      return true;
+    if (obj == null)
+      return false;
+    if (getClass() != obj.getClass())
+      return false;
+    ComparableEvent other = (ComparableEvent) obj;
+    if (outcome != other.outcome)
+      return false;
+    if (!Arrays.equals(predIndexes, other.predIndexes))
+      return false;
+    if (seen != other.seen)
+      return false;
+    if (!Arrays.equals(values, other.values))
+      return false;
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
We use a different style for hashCode in this project, Java now contains via Objects utils to do this nicer.